### PR TITLE
The file `CMakeTests.cmake` is no longer included if `ENABLE_TESTING` is off

### PR DESCRIPTION
**Pull request by isaacbrodsky**
This is a quick solution for uber/h3-py#239, but does not address the underlying issue of the file not being found. I do think this is useful because it allows us to unindent the `CMakeTests.cmake` file.

Also fixed a missing CHANGELOG item for #603

### CODE REVIEW

**Review by ajfriend — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ The public API of this library consists of the functions declared in file
 [h3api.h.in](./src/h3lib/include/h3api.h.in).
 
 ## [Unreleased]
+### Fixed
+- Fix polyfill bug when vertex latitude exactly matches cell center. (#603)
```
Comment (nrabinowitz) [inline CHANGELOG.md:10]: 🙏

File: CMakeTests.cmake [hunk 2]
```diff
@@ -14,215 +14,213 @@
 
 # Test code for H3
 
-if(H3_IS_ROOT_PROJECT AND BUILD_TESTING)
-    option(BUILD_ALLOC_TESTS "Build tests for custom allocation functions" ON)
-    option(PRINT_TEST_FILES "Print which test files correspond to which tests" OFF)
+option(BUILD_ALLOC_TESTS "Build tests for custom allocation functions" ON)
+option(PRINT_TEST_FILES "Print which test files correspond to which tests" OFF)
+
+include(TestWrapValgrind)
```
Comment (nrabinowitz) [inline CMakeTests.cmake:20]: The visual diff here is lousy, but I assume from here on down it's just indentation changes?

**Review by nrabinowitz — APPROVED** [review]

File: CMakeTests.cmake (hunk 2 above)
Comment (isaacbrodsky) [reply CMakeTests.cmake:20]: Correct, there should be no substantive changes to this file. Reindenting in VS Code and comparing to master I see only a couple lines where indents differ but no other changes.

Comment (dfellis) [reply CMakeTests.cmake:20]: You can tell github diff to ignore indentation and then it clears up here.

Comment (dfellis) [reply CMakeTests.cmake:20]: Screenshot from 2022-05-25 19-48-58
Screenshot from 2022-05-25 19-49-11

**Pull request merged**